### DISCUSSION TITLE
Official Nvidia Support for Yocto?

### DISCUSSION THREAD

**Discussion by philipp-schmidt** [OE4T/meta-tegra#1219] [General]
Just discovered this repo, and yet again it is wonderful to see what the open source community is capable of. I consider this Yocto support for Jetsons a necessary step for the effective deployment of Jetsons in commercial projects - yet it is competely backed by a few very capable people doing this most likely in their free time. While this is as always very much appreciated - what is Nvidias stance on this? Is there an official endorsement / statement from Nvidia? Where do the maintainers of this repo go to when they have questions/issues with the newest L4T release? Will Nvidia make sure future changes to L4T will stay compatible with the way this repo supports Yocto? With Nvidia also benefiting from this, I was wondering if there was ever a discussion for them to support this repo in a more official capacity?

**Comment by ichergui** [comment]
Welcome @philipp-schmidt to the OE4T community. 
Thanks for the nice message. We are working closely with Nvidia team and they are supporting us. 
FYI, we have our monthly meetings, you can join to get updates regarding the jetson platforms and the ongoing work.
Also, you can join the [gitter channel](https://github.com/OE4T/meta-tegra/wiki) to get updates or ask any question regarding OE4T 
I will let @madisongh @dwalkes @kekiefer adding comment here.

**Comment by madisongh** [comment]
I can't speak for NVIDIA, and I don't have any official statement I can point to, but my understanding is that they will continue to use Ubuntu as the base for Jetson Linux (aka L4T), as they have done for many years now.  Yocto is supported through their [Jetson ecosystem partners](https://developer.nvidia.com/embedded/ecosystem?specialty=yocto).

But as Ilies mentions above, we do communicate with the Jetson Linux folks at NVIDIA, and they have provided us with early access kits for major L4T updates, as well as some pre-release hardware (such as Orin NX and Orin Nano units) to enable us to keep the layer up-to-date in a timely fashion.

**Comment by ibstewart** [comment]
While NVIDIA has not made any official support promises with respect to Yocto and `meta-tegra`, the [NVIDIA Holoscan](https://github.com/nvidia-holoscan) team is providing an additional metadata layer that adds onto `meta-tegra`, [meta-tegra-holoscan](https://github.com/nvidia-holoscan/meta-tegra-holoscan), in order to support Holoscan developer kits like the [IGX Orin](https://www.nvidia.com/en-us/edge-computing/products/igx/) and to provide additional dependencies required by the [Holoscan SDK](https://github.com/nvidia-holoscan/holoscan-sdk). This support for the Holoscan developer kits includes some additions like the ability to use a discrete GPU (such as the A6000 available with the IGX Orin devkit), Mellanox OFED drivers, as well as some support for 3rd party peripherals such as video I/O boards offered by NVIDIA partners. The goal of this is to get Holoscan customers  -- some of which have never worked with Yocto before -- up and running with a reference platform that could ultimately be tailored to become a production-ready product.

With that said, this additional NVIDIA Holoscan support wouldn't exist without the amazing work that the OE4T community has done and continues to do, and NVIDIA certainly recognizes and appreciates this. The plan for the NVIDIA Holoscan OE support is to only keep Holoscan-specific parts in `meta-tegra-holoscan`, upstreaming any general patches or work into `meta-tegra` whenever possible so that the Holoscan layer remains an add-on to `meta-tegra`. But so far the reality has been than any time something has been needed for Holoscan, a fetch from `meta-tegra` usually pulls the patches from the OE4T community that already has it covered!
